src/ParseError.js

```diff
@@ -66,6 +71,8 @@ class ParseError {
         self.__proto__ = ParseError.prototype;
         // $FlowFixMe
         self.position = start;
+        self.length = end - start;
```

Comment:
It occurs to me that this computes `NaN` if `start` and `end` are undefined. Indeed, Flow caught this as well. Perhaps this line should be wrapped in `if (start != null && end != null)`?

Apparently we also need `// $FlowFixMe` because `self` is typed as `Error` here. But maybe this could be fixed via type casting? (In TypeScript it'd be `as`.)

(You can run `yarn test` to run Lint, Jest, and Flow yourself.)

Reply:
Ah okay, I always just used `yarn test:jest`, makes more sense now 😅

Reply:
I tried to cast this error, but that does not seem to help. I get syntax errors, so I assume that this is not TypeScript?
Sorry, I am super unexperienced with JS.

Using `// $FlowFixMe` works fine.

Reply:
Thanks for the hint! However this all does not appear to work. As far as I can see, `ParseError` and `Error` are distinct classes, having nothing to do with each other. So if I cast `self` to `ParseError`, I just get errors when returning `self` from the constructor, which expects `Error`. So it's just the other way around. I followed the StackOverflow link at the top of the class declaration. The top answer provides an alternative available since ES6 by just extending `Error`. This would allow to remove some of the hackery around `ParseError`.

However, lint is unhappy with this, because extending classes appears to be prohibited in the `eslintrc` file (`ClassDeclaration[superClass]`).

If I remove this restriction, the tests appear to run fine on my machine 😅